### PR TITLE
mplab-xc16: update livecheck

### PR DESCRIPTION
**Pull request by samford**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

-----

As described in https://github.com/Homebrew/homebrew-cask/pull/163501, the existing livecheck block redirects to https://www.microchip.com/en-us/tools-resources/develop/mplab-xc-compilers but returns an `Unable to get versions` error, since the page doesn't contain a link to the installer file. This resolves the issue by updating the `livecheck` block to check the "MPLAB XC16 Compiler" page (linked from the existing page as "View XC16 Compiler Downloads"), which links to the cask file.

A 2.10 version is available but my local machine isn't supported (macOS Sonoma, ARM64), so I can't debug the install to sort it out. There are a couple previous PRs that attempted to update this to 2.10 (#145808, #154311), so it seems like there are some issues that need to be worked out. I've pared this PR down to just the `livecheck` block change, to allow it to move forward.

### CODE REVIEW

**Comment by samford** [comment]
Labeling this syntax-only, simply to avoid the CI failure due to the cask being 2.09 and the latest livecheck version being 2.10.

**Comment by SMillerDev** [comment]
Do casks actually fail on that? I thought they didn't

**Comment by samford** [comment]
> Do casks actually fail on that? I thought they didn't

You may be thinking of homebrew/core CI, where a `livecheck` issue doesn't (usually) cause CI to fail. A cask/livecheck version mismatch is a failure on cask CI, so the cask version and/or the `livecheck` block generally has to be updated to pass CI. Granted, that's my understanding from my limited experience with casks and the cask maintainers may have a better explanation.

**Comment by SMillerDev** [comment]
Nevermind, I misread the comment. But why not update the cask then.

**Comment by samford** [comment]
> why not update the cask

That's usually how I would approach it and this PR originally bumped the version as well. However, it's not clear how to fix the cask for 2.10 (there were two previous PRs that attempted to update to 2.10 but were closed) and I can't install it locally to figure it out (the install script exits with `The current OS X version is not supported` on ARM64). There are a ton of permission issues when trying to install the cask on CI (not sure if it's related to needing `sudo`), so it didn't seem like I could use CI to iteratively modify this until it works.

Basically, it's an unusual scenario and I figured it would be better to simply fix the `livecheck` block (as a follow-up to #163501) and leave the upgrade to someone with an Intel machine and a better understanding of what this cask is doing.

**Comment by krehel** [comment]
Agreed.  Thanks @samford.  One day we'll have to figure out how to get this upgraded.

**Review by krehel — APPROVED** [review]
